src/main/java/com/revature/services/EvalLogicImpl.java

```diff
@@ -17,34 +15,67 @@
 
 	@Autowired
 	private EvalRepository dao;
+	
+
+//CREATE----------------------------
 
+//RETRIEVE--------------------------
 	@Override
 	public Eval getEvalById(int id) {
-		return dao.findOne(id);
-
+		Eval eval = dao.findOne(id);
+		return eval;
 	}
 	
 	@Override
-	public List<Eval> getEvalsByBatch(int id){
-		
-		return dao.findAllByBatchId(id);
+	public Page<Eval> getEvalsByBatch(Pageable pageable, int id, String evalParam){
+		Page<Eval> evals;
+		if (evalParam.equalsIgnoreCase("both")) {
+			evals = dao.findAllByBatchId(pageable, id);
+			System.out.println(evals);
+		} else {
+			evals = dao.findAllByBatchIdAndEvalTypeDescriptionIgnoreCase(pageable, id, evalParam);
+			System.out.println(evals);
+		}
+		System.out.println(evals);
+		return evals;
 	}
 	
 	@Override
-	public List<Eval> getEvalsByWeek(int id, int num){
-		return dao.findAllByWeek(id,num);
-
+	public Page<Eval> getEvalsByWeek(Pageable pageable, int id, int num, String evalParam){
+		Page<Eval> evals;
+		if (evalParam.equalsIgnoreCase("both")) {
+			evals = dao.findAllByBatchIdAndWeek(pageable, id, num);
+		} else {
+			evals = dao.findAllByBatchIdAndWeekAndEvalTypeDescriptionIgnoreCase(pageable, id, num, evalParam);
+		}
+		System.out.println(evals);
+		return evals;
 	}
-
+	
 	@Override
-	public List<Eval> getEvalsByPerson(int id) {
-		return dao.findAllByTraineeId(id);
-
+	public Page<Eval> getEvalsByPerson(Pageable pageable, int id, String evalParam) {
+		Page<Eval> evals;
+		if (evalParam.equalsIgnoreCase("both")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Move the "both" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1132)